doc/api/crypto.md

```diff
@@ -2651,6 +2651,18 @@ added: v10.0.0
 Enables the FIPS compliant crypto provider in a FIPS-enabled Node.js build.
 Throws an error if FIPS mode is not available.
 
+### crypto.signOneShot(data, privateKey)
+<!-- YAML
+added: REPLACEME
+-->
+* `data` {Buffer | TypedArray | DataView}
+* `privateKey` {Buffer | KeyObject}
+* Returns: {Buffer}
+
+Calculates and returns the signature for `data` using the given private key.
+
+This method only supports Ed25519 and Ed448 keys.
```

Comment:
Is there a reason for this restriction? I think having these "one shot" implementations could solve problems such as #25857 where JS object creation time is critical.

Reply:
Currently the reason is to keep things simple and the fact that EdDSA does not utilize a hash algorithm argument with OpenSSL's API.

Reply:
> The lack of a digest arg and signing algorithm makes this very EdDSA specific, if its going to be that specific, might as well just call it what it is: `crypto.eddsaSign()`

There was interest shown earlier about possibly allowing an algorithm in the future somehow, so I made the name more generic. I honestly don't care so much about the name as long as it's fairly accurate and concise and doesn't look like it came from Java.

> My skim of the RFCs gave me the impression that there was PureEdDSA and EdDSA, but this API doesn't allow a choice between them. Not a blocker, but I'm curious, does OpenSSL not support them, or did I misunderstand?

As noted in the referenced issue, OpenSSL implements PureEdDSA which does not support streaming data.

> but using an options object might be nicer, particularly since all 5 args can be `string`, so its not easy to just not provide them and have the API provide default values.

Possibly, but the other reason for using an object is that having the encoding parameters be optional like they are in the streaming API would be very hard to detect all of the various possibly-string arguments. As noted in other comments, I did not bother implementing the encoding parameters because they are part of a legacy API and using `Buffer.from()` and `buffer.toString()` are trivial these days and it greatly simplifies the implementation.

Reply:
OK, so that addresses most of my comments, but the plan for this is unclear:

> There was interest shown earlier about possibly allowing an algorithm in the future somehow, so I made the name more generic. I honestly don't care so much about the name as long as it's fairly accurate and concise and doesn't look like it came from Java.

I don't know what Java's API conventions are, so this doesn't illuminate the kinds of names you like, for me at least.

I still don't love `xxxOneShot()` as a naming convention, but I can live with it.

I don't think the API should have a generic name if it doesn't have enough args to one-shot a generic sign algorithm. `digest` should be added as the first arg, IMO, if the name stays generic.